lib/core/message.toit

```diff
@@ -88,23 +93,82 @@ process_messages_:
       // task's deadline if any.
       critical_do --no-respect_deadline:
         if message is Array_:
-          type ::= message[0]
-          handler ::= system_message_handlers_.get type --if_absent=:
-            print_ "WARNING: unhandled system message $type"
-            continue
-          gid ::= message[1]
-          pid ::= message[2]
-          arguments ::= message[3]
-          message = null  // Allow garbage collector to free.
-
-          handler.on_message type gid pid arguments
+          entered := false
+          lambda := ::
+            entered = true
+            type ::= message[0]
+            handler ::= system_message_handlers_.get type
+            if handler:
+              gid ::= message[1]
+              pid ::= message[2]
+              arguments ::= message[3]
+              message = null  // Allow garbage collector to free.
+              handler.on_message type gid pid arguments
+            else:
+              print_ "WARNING: unhandled system message $type"
+
+          reused := true
+          processor := message_processor_
+          if not processor.run lambda:
+            processor = MessageProcessor_ lambda
+            message_processor_ = processor
+            reused = false
+          processor.detach_if_necessary_
+          if not entered:
```

Comment:
Will remove this code before merging. It is here to make any issue with this stand out more during testing.